test/tools/ossfuzz/protoToAbiV2.h

```diff
@@ -0,0 +1,249 @@
+	void writeHelperFunctions();
+	void bufferVarDef(std::string const& _varName, std::string const& _rhs);
+
+	// Inline functions
+	inline unsigned getNextCounter()
+	{
+		return m_counter++;
+	}
+
+	inline std::string newVarName()
+	{
+		return  ("x_" + std::to_string(m_varCounter++));
+	}
+
+	inline std::string bytesArrayValueAsString()
+	{
+		return ("\"" + u256(h256(std::to_string(getNextCounter()))).str() + "\"");
+	}
+
+	inline std::string getQualifier(dataType _dataType)
+	{
+		return (!(isValueType(_dataType) || m_isStateVar) ? "memory" : "");
+	}
+
+	// Static declarations
+	static std::string structTypeAsString(StructType const& _x);
+	static std::string intValueAsString(unsigned _width, unsigned _counter);
+	static std::string uintValueAsString(unsigned _width, unsigned _counter);
+	static std::string integerValueAsString(bool _sign, unsigned _width, unsigned _counter);
+	static std::string addressValueAsString(unsigned _counter);
+	static std::string fixedByteValueAsString(unsigned _width, unsigned _counter);
+	static std::vector<std::pair<bool, unsigned>> arrayDimensionsAsPairVector(ArrayType const& _x);
+	static std::string arrayDimInfoAsString(ArrayDimensionInfo const& _x);
+	static void arrayDimensionsAsStringVector(
+		ArrayType const& _x,
+		std::vector<std::string>&
+	);
+	static std::string bytesArrayTypeAsString(DynamicByteArrayType const& _x);
+	static std::string arrayTypeAsString(std::string const&, ArrayType const&);
+	static std::string delimiterToString(delimiter _delimiter);
+
+	// Static inline functions
+	static inline bool isValueType(dataType _dataType)
+	{
+		return (_dataType == dataType::VALUE);
+	}
+
+	static inline unsigned getIntWidth(IntegerType const& _x)
+	{
+		return (8 * ((_x.width() % 32) + 1));
+	}
+
+	static inline bool isIntSigned(IntegerType const& _x)
+	{
+		return _x.is_signed();
+	}
+
+	static inline std::string getIntTypeAsString(IntegerType const& _x)
+	{
+		return ((isIntSigned(_x) ? "int" : "uint") + std::to_string(getIntWidth(_x)));
+	}
+
+	static inline unsigned getFixedByteWidth(FixedByteType const& _x)
+	{
+		return ((_x.width() % 32) + 1);
+	}
+
+	static inline std::string getFixedByteTypeAsString(FixedByteType const& _x)
+	{
+		return ("bytes" + std::to_string(getFixedByteWidth(_x)));
+	}
+
+	static inline std::string getAddressTypeAsString(AddressType const& _x)
+	{
+		return (_x.payable() ? "address payable": "address");
+	}
+
+	// TODO: Fix hashing function
+	static inline u256 hashUnsignedInt(unsigned _counter)
+	{
+        return h256(std::to_string(_counter), h256::FromBinary, h256::AlignLeft);
+	}
+
+	static inline u256 maskUnsignedInt(unsigned _counter, unsigned _numMaskOctets)
+	{
+//	  return hashUnsignedInt(_counter) & u256("0x" + std::string(_numMaskOctets, 'f'));
+	  return _counter & u256("0x" + std::string(_numMaskOctets, 'f'));
+	}
+
+	// Requires caller to pass number of octets (twice the number of bytes) as second argument.
```

Comment:
Ah! But isn't octet just a synonym for byte?

Reply:
You are right, I think what I meant is nibble :-)